tox.ini

```diff
@@ -1,20 +1,100 @@
 [tox]
-envlist = py38, py39, py3.10, py311, py312, pypy3
+envlist = pypy3,py3{9,10,11,12,13}-{linux,macos,windows}
 skip_missing_interpreters = true
+isolated_build = true
+skipsdist = true
 
 [gh-actions]
 python =
-    3.8: py38
     3.9: py39
     3.10: py310
     3.11: py311
     3.12: py312
-    pypy3.10: pypy3
+    3.13: py313
+
+[gh-actions:env]
+PLATFORM =
+    ubuntu-latest: linux
+    ubuntu-22.04: linux
+    ubuntu-24.04-arm: linux
+    macos-latest: macos
+    macos-15: macos
+    windows-latest: windows
+    windows-2025: windows
 
 [testenv]
-extras = dev
+skip_install = true
+
+passenv =
+    DISPLAY
+    XAUTHORITY
+    HOME
+    USERNAME
+    USER
+    XDG_*
+    CI
+    OS
+    PYTHONIOENCODING
+    PIP_DOWNLOAD_CACHE
+
+setenv =
+    PYTHONPATH = {toxinidir}
+
+allowlist_externals =
+    bash
+
+deps =
+    -e .[test,dev]
```

Comment:
Is there an advantage for this over "extras"?

Reply:
Visibility? otherwise it's just `-e .`

Reply:
Does that change anything? AFAIK tox sandboxes the environment anyways.